### PR TITLE
Add popup modal

### PR DESCRIPTION
**Pull request by jiejasonliu**
just gonna document everything I did here, perhaps a little too verbose but the more the merrier 🤷 

- new popup modal component
  - **(lifted state):** consumer of modal handles `showTrigger` and `onClose`  
  - **(optional prop):** close modal when clicking outside of the modal contents
  - new `useFocusTrap` hook to force accessibility navigation (`Tab` key) inside a parent ref
    - can repurpose/use this to select specific elements when tabbing in the `meta-data-editor`
  - new `useFocusFirst` hook to auto-focus the first element inside a parent ref
  - use React portals to render modal outside of parent DOM while still propagating events
    - **advantages:** less style conflicts (z-index, absolute styling), less rerenders
- change `cancel-icon.svg` to `CancelIcon` component

https://user-images.githubusercontent.com/29434693/162556016-02c69329-a96e-43f3-b289-9c8e7c0472e7.mp4

### CODE REVIEW

File: src/components/popup-modal/popup-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  useOutsideClick(contentRef, () => onClose(), outsideClickFiresOnClose);
+
+  // trap accessibility controls (i.e. tabbing) in the content
+  useFocusTrap(contentRef);
+
+  // accessibility: auto-focus the first focusable element (if any)
+  useFocusFirst(contentRef, showTrigger);
+
+  // render onto <div id="portal"></div> instead of in parent hierarchy but still propagate events
+  return ReactDOM.createPortal(
```
Comment (derricklee43) [inline src/components/popup-modal/popup-modal.tsx:35]: Damn!! This is some black magic type shit! That's so dope!

**Comment by derricklee43** [comment]
That was fucking quick! I'm super impressed by the hooks and portals, did you just pick that up??

**Review by derricklee43 — APPROVED** [review]
LGTM!

**Comment by jiejasonliu** [comment]
> That was fucking quick! I'm super impressed by the hooks and portals, did you just pick that up??

haha thanks! portals were mainly inspired by a lot of medium articles and hooks from open source modal components

**Pull request merged**